tempesta_fw/procfs.c

```diff
@@ -0,0 +1,180 @@
+#include <linux/proc_fs.h>
+#include <linux/seq_file.h>
+
+#include "procfs.h"
+
+DEFINE_PER_CPU_ALIGNED(tfw_perfstat_t, tfw_perfstat);
+
+static struct proc_dir_entry *tfw_procfs_tempesta;
+static struct proc_dir_entry *tfw_procfs_perfstat;
+
+void
+tfw_perfstat_collect(tfw_perfstat_t *stat)
+{
+	int cpu;
+
+	preempt_disable();
```

Comment:
Why do you disable interrupts? Typically there is nothing serious to show bit inconsistent statistic